tests/models/imagebind/test_processor_imagebind.py

```diff
@@ -0,0 +1,464 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import shutil
+import tempfile
+import unittest
+
+import numpy as np
+import pytest
+
+from transformers import CLIPTokenizer, CLIPTokenizerFast, ImageBindFeatureExtractor
+from transformers.testing_utils import require_torch, require_torchaudio, require_vision
+from transformers.utils import is_vision_available
+
+
+if is_vision_available():
+    from PIL import Image
+
+    from transformers import ImageBindImageProcessor, ImageBindProcessor
+
+from ...test_processing_common import ProcessorTesterMixin
+
+
+@require_vision
+@require_torchaudio
+class ImageBindProcessorTest(ProcessorTesterMixin, unittest.TestCase):
+    processor_class = ImageBindProcessor
+
+    def setUp(self):
+        self.tmpdirname = tempfile.mkdtemp()
+        self.checkpoint = "EduardoPacheco/imagebind-huge"
+
+        image_processor = ImageBindImageProcessor()
+        tokenizer_slow = CLIPTokenizer.from_pretrained(self.checkpoint)
+        tokenizer_fast = CLIPTokenizerFast.from_pretrained(self.checkpoint)
+        feature_extractor = ImageBindFeatureExtractor()
+
+        processor_slow = ImageBindProcessor(image_processor, tokenizer_slow, feature_extractor)
+        processor_fast = ImageBindProcessor(image_processor, tokenizer_fast, feature_extractor)
+
+        processor_slow.save_pretrained(self.tmpdirname)
+        processor_fast.save_pretrained(self.tmpdirname)
+
+    def get_tokenizer(self, **kwargs):
+        return CLIPTokenizer.from_pretrained(self.checkpoint, **kwargs)
+
+    def get_rust_tokenizer(self, **kwargs):
+        return CLIPTokenizerFast.from_pretrained(self.checkpoint, **kwargs)
+
+    def get_image_processor(self, **kwargs):
+        return ImageBindImageProcessor.from_pretrained(self.checkpoint, **kwargs)
+
+    def get_feature_extractor(self, **kwargs):
+        return ImageBindFeatureExtractor.from_pretrained(self.checkpoint, **kwargs)
+
+    def tearDown(self):
+        shutil.rmtree(self.tmpdirname)
+
+    def prepare_image_inputs(self):
+        """This function prepares a list of PIL images, or a list of numpy arrays if one specifies numpify=True,
+        or a list of PyTorch tensors if one specifies torchify=True.
+        """
+
+        image_inputs = [np.random.randint(255, size=(3, 30, 400), dtype=np.uint8)]
+
+        image_inputs = [Image.fromarray(np.moveaxis(x, 0, -1)) for x in image_inputs]
+
+        return image_inputs
```

Comment:
I think this one is factored into the Mixin now

Reply:
yes, thank you for noticing, removed it :+1: